lmcache/address_manager/disk_address_manager.py

```diff
@@ -0,0 +1,285 @@
+
+        self.update_lock = threading.Lock()
+
+        self.evictor = LRUEvictor()
+
+    def _key_to_path(
+        self,
+        key: CacheEngineKey,
+    ) -> str:
+        """
+        Convert key to path_name
+
+        Input:
+            key: the key of the token chunk, including prefix hash and format
+
+        Returns:
+            returns the path name
+        """
+        return self.path + key.to_string().replace("/", "-") + ".pt"
+
+    def remove(
+        self,
+        key: CacheEngineKey,
+    ) -> None:
+        """
+        Remove the KV cache chunk by the given key
+
+        Input:
+            key: the key of the token chunk, including prefix hash and format
+
+        """
+
+        self.update_lock.acquire()
+        path = self.dict[key].path
+        self.dict.pop(key)
+        self.update_lock.release()
+
+        os.remove(path)
+
+    def contains(
+        self,
+        key_str: str,
+    ) -> str:
+        """
+        Check if the cache engine contains the key.
+
+        Input:
+            key: the key of the token chunk, including prefix hash and format
+
+        Returns:
+            path if the cache engine contains the key, "" otherwise
+        """
+        key = CacheEngineKey.from_string(key_str)
+        with self.update_lock:
+            return self.dict[key].path if key in self.dict else ""
+
+    def batched_contains(
+        self,
+        keys_strs: Iterable[str],
+    ) -> Iterable[str]:
+        return [self.contains(key_str) for key_str in keys_strs]
+
+    def write_check(self, key_str: str, kv_size: float) -> str:
+        """
+        The logic here is that:
+            return "" if the key is already in the cache, 
+                so that backend will not write it again
+            return path if the key is not in the cache, 
+                so that backend will write it to the path
+        if path is not in the cache, initialize the cache with ("",0)
+        """
+        key = CacheEngineKey.from_string(key_str)
+        with self.update_lock:
+            if key in self.dict:
+                return ""
+
+            evict_keys, put_status = self.evictor.update_on_put(
+                self.dict, int(kv_size * (1024 * 1024)))
+
+        # Abort put if cache too big
+        if put_status == PutStatus.ILLEGAL:
+            return ""
+
+        # evict caches
+        for evict_key in evict_keys:
+            self.remove(evict_key)
+
+        with self.update_lock:
+            self.dict[key] = DiskCacheMetadata("", 0)
+
+        return self._key_to_path(key)
+
+    def batched_write_check(self, key_strs: Iterable[str],
+                            kv_size: float) -> Iterable[str]:
+        paths = []
+        kv_size = int(kv_size * (1024 * 1024))
+
+        self.update_lock.acquire()
+        for key_str in key_strs:
+            key = CacheEngineKey.from_string(key_str)
+            if key in self.dict:
+                kv_size = kv_size - self.dict[key].size
```

Comment:
Why doing `kv_size = kv_size - self.dict[key].size` in `batched_write_check` but not in `write_check`?

Reply:
Because write_check will return if it's in the dict

Reply:
```
kv_size: the size of the KV Cache chunk group in GB, 
            free up the space if the existing cache is too big
            
# Do not free up space for existed KV Cache chunks
```
Just see the comment

Reply:
For write_check, if key in self.dict, the function will return "" and not incur evictor.